### ISSUE TITLE
Aircraft approach & autoland causes plane crash

### ISSUE THREAD

**Issue by morphelianus** [MuMech/MechJeb2#1312]
#798  MechJeb Version
2.10.0.0

## KSP Version
Stock 1.10.0.2917 de-de
Making History 1.10.0
Breaking Ground 1.5.0

## Description
Whenever I enable the Aircraft Approach & Autoland the planes crashes instandly as MechJeb sets the Target Altitude to -212608 meter. This bug occurs since MechJeb 2.10.0.0

## Replication Case
This bug happens with each one of my planes and is not deticated to a single design.

## KSP.log
[KSP_x64.gc.log](https://github.com/MuMech/MechJeb2/files/4887351/KSP_x64.gc.log)
[KSP.log](https://github.com/MuMech/MechJeb2/files/4887352/KSP.log)

**Comment by felixscheffer** [comment]
I am working on a fix.

**Comment by morphelianus** [comment]
After some test for the other bug I submitted I was able to reproduce the error more precisely. Whenever I engage the autopilot above 1000 meters everything is fine but below it sets the target altitude for the first nav point somewhere deep below base ground. I hope this helps a bit more. In case you need more details just leave a comment.

**Comment by felixscheffer** [comment]
Thanks for the update.

I created a PR to fix this issue, so once the PR is merged it should be working again (in the dev build for now). I was using the wrong KSP method to convert the altitude back to a vector. This should only affect cases where the aircraft is below the glideslope when you engage autoland . If you start your descent above the glideslope, everything should be fine.

**Comment by morphelianus** [comment]
## MechJeb Version (DEV RELEASE)
2.10.0.0-1001

Engaging aircraft approach and autoland below the slope sets the target altitude now correctly.

However, I recognized a short drop of the current vertical speed whenever I engaged aircraft approach and autoland after I engaged aircraft autopilot. Sometimes it could even drop into negative values. After a couple of seconds the current vertical speed matches again the target vertical speed and everything is fine.

@felixscheffer 
Shall I open a new issue and record a video?

**Comment by felixscheffer** [comment]
> @felixscheffer
> Shall I open a new issue?

Yes please

**Comment by morphelianus** [comment]
> > @felixscheffer
> > Shall I open a new issue?
> 
> Yes please

Done. Nevertheless, this issue is solved. Well done.
